solr/core/src/java/org/apache/solr/core/ZkContainer.java

```diff
@@ -64,7 +64,7 @@ public class ZkContainer {
   private SolrZkServer zkServer;
 
   private ExecutorService coreZkRegister =
-      ExecutorUtil.newMDCAwareCachedThreadPool(new SolrNamedThreadFactory("coreZkRegister"));
+      ExecutorUtil.newMDCAwareCachedThreadPool(Integer.getInteger("Zk.core.register.threads", Integer.MAX_VALUE), new SolrNamedThreadFactory("coreZkRegister"));
```

Comment:
Is `zk.core.register.threads` a new property? And its significance wrt the main goal of this PR is that now `coreZkRegister` executor has potentially a lot more heavy work to do (picking up all the `initSearcher()` that was dropped from SolrCore ctor)?

Reply:
I think we don't need this property. Likely we need some mechanism to make n-parallel calls for openSearcher.